feat: ✨ add page template

limwa opened the pull request:
This pull request's main objective is to create a template that can be used by our tool, along with Jinja (most likely), to create a page for a given quiz.

_This is still a work in progress._

limwa committed:
feat: :alembic: add example quiz page

This commit introduces an example of what this tool will be able to generate.
It must be noted, though, that this is obviously a prototype and will be changed in the future.

limwa commented:
Click [here](https://htmlpreview.github.io/?https://github.com/MIEIC-FEUP/quiz-trainer/blob/feature/create-page-template/examples/quiz.html) to see a preview of the example page!

gwealm commented:
I've noticed that the button to clear the answers only supports a full clear. What if we create an extra option on each question to clear the current answer? What do you think?

limwa commented:
Good point! I'm thinking we put a danger button after the options for the answer. However, if we do that, I think we will need to handle the clear in JavaScript.

limwa committed:
feat: :sparkles: add clear button to every question

limwa committed:
feat: :sparkles: add fieldset reset functionality

limwa committed:
feat: :sparkles: add answer checking to example page

limwa committed:
build(deps): :heavy_plus_sign: add Jinja2 to dependencies

limwa committed:
feat: :alembic: preliminary support for fill button

This commit adds fill button support for radio inputs.
Later on, other types of inputs will be supported as well.

This commit also changes the behaviour of clear buttons, in order to fix some unwanted behaviour.

limwa committed:
refactor: :recycle: merge all scripts into a single one

limwa committed:
feat: :heavy_plus_sign: add mathjax

limwa committed:
style: :lipstick: use more expressive labels for buttons

limwa committed:
style: :lipstick: make aside element responsive

limwa committed:
refactor: :fire: remove weird hack using 2 h1 elements

limwa committed:
refactor: :fire: remove unnecessary selectors

limwa committed:
fix: :lipstick: width of elements not being applied

limwa committed:
style: :lipstick: make navigation more intuitive on small screens

limwa committed:
feat: :lipstick: provide better-looking feedback for radio elements

limwa committed:
revert(898006a): remove better feedback for answers

This doesn't make sense as the check button is only meant to check the answers.
If the user wants to see the correct answer, there is a fill button available.

limwa committed:
feat: :sparkles: implement other types of question types

limwa commented:
The only thing left now is navigation!

limwa committed:
feat: :sparkles: add navigation buttons

limwa committed:
style: :lipstick: make some finishing touches

In this commit, the side of the navigation widget was reduced by 1rem, making it a little bit more spaced.
The name of the navigation widget was also changed to the name of the test.
Finally, keyboard navigation in the navigation widget was improved.

limwa committed:
refactor: put meaningful questions in example page

limwa commented:
Navigation is done!

limwa committed:
fix: :bug: two nav buttons have the same question number

limwa commented:
oi
